sheets436/src/main/java/edu/umd/cmsc436/sheets/WriteDataTask.java

```diff
@@ -133,6 +137,35 @@ protected Exception doInBackground(WriteData... params) {
                 } else {
                     this.writeToPrivate(wd);
                 }
+            } catch (com.google.api.client.googleapis.json.GoogleJsonResponseException e) {
+                if (e.getDetails().getErrors().get(0).getMessage().contains("Unable to parse range:")) {
+                    // Create request to add a new sheet
+                    AddSheetRequest addReq = new AddSheetRequest();
+                    addReq.setProperties(new SheetProperties().setTitle(
+                            wd.testType.toId().substring(1, wd.testType.toId().length() - 1)));
+                    // A bunch of stupid shit for sheets API request building
+                    Request req = new Request();
+                    req.setAddSheet(addReq);
+                    ArrayList<Request> reqList = new ArrayList<>();
+                    reqList.add(req);
+                    BatchUpdateSpreadsheetRequest batchReq = new BatchUpdateSpreadsheetRequest();
+                    batchReq.setRequests(reqList);
+                    try {
+                        // Update the request
+                        sheetsService.spreadsheets().batchUpdate(spreadsheetId, batchReq).execute();
+                        // Retry our original write
+                        if (wd.central) {
+                            this.writeToCentral(wd);
+                        } else {
+                            this.writeToPrivate(wd);
+                        }
+                    } catch (Exception e2) {
```

Comment:
It looks kinda like callback hell, except with try-catch. Is there any way you can refactor to make it cleaner (e.g. make the try blocks tighter)? When I was thinking about having a small, single layer try-catch around each call to `execute`.